pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -180,6 +188,17 @@ func (c *Controller) Apply(change *sink.Change) error {
 				conf.Name: conf,
 			}
 		}
+
+		_, err := c.ledger.Put(conf.Key(), obj.Metadata.Version)
+		if err != nil {
+			log.Warnf(ledgerLogf, err)
+		}
+	}
+	for _, removed := range change.Removed {
+		err := c.ledger.Delete(kube.KeyFunc(change.Collection, removed))
+		if err != nil {
+			log.Warnf(ledgerLogf, err)
+		}
```

Comment:
no locks needed? ads code could be reading all of them right?

Reply:
The ledger is thread safe.  No locks needed.